examples/schedulers/static_thread_pool.hpp

```diff
@@ -88,11 +89,292 @@ namespace example {
+                for (Shape i = begin; i < end; ++i) {
+                  self.fn_(i, args...);
+                }
+              };
+
+              auto completion = [&](auto&... args) {
+                std::execution::set_value((Receiver&&)self.receiver_, std::move(args)...);
+              };
+
+              if constexpr (MayThrow) {
+                try {
+                  self.apply(computation);
+                } catch(...) {
+                  std::uint32_t expected = total_threads;
+
+                  if (self.thread_with_exception_.compare_exchange_strong(
+                          expected, tid, 
+                          std::memory_order_relaxed,
+                          std::memory_order_relaxed)) {
+                    self.exception_ = std::current_exception();
+                  }
+                }
+
+                const bool is_last_thread = self.finished_threads_.fetch_add(1) == (total_threads - 1);
+
+                if (is_last_thread) {
+                  if (self.exception_) {
+                    std::execution::set_error((Receiver&&)self.receiver_, self.exception_);
+                  } else {
+                    self.apply(completion);
+                  }
+                }
+              } else {
+                self.apply(computation);
+
+                const bool is_last_thread = self.finished_threads_.fetch_add(1) == (total_threads - 1);
+
+                if (is_last_thread) {
+                  self.apply(completion);
+                }
+              }
+            };
+          }
+        };
+
+      template <class SenderId, class ReceiverId, class Shape, class Fn, bool MayThrow>
+        struct bulk_receiver {
+          using Sender = std::__t<SenderId>;
+          using Receiver = std::__t<ReceiverId>;
+
+          using shared_state = bulk_shared_state<SenderId, ReceiverId, Shape, Fn, MayThrow>;
+
+          shared_state& shared_state_;
+
+          template <class... As>
+          friend void tag_invoke(std::execution::set_value_t, bulk_receiver&& self, As&&... as) noexcept {
+            using tuple_t = std::execution::__decayed_tuple<As...>;
+
+            shared_state& state = self.shared_state_;
+
+            if constexpr (MayThrow) {
+              try {
+                state.data_.template emplace<tuple_t>((As &&) as...);
+              } catch (...) {
+                std::execution::set_error((Receiver&&)state.receiver_, std::current_exception());
+              }
+            } else {
+              state.data_.template emplace<tuple_t>((As &&) as...);
+            }
+
+            if (state.shape_) {
+              state.pool_.bulk_enqueue(&state, state.num_agents_required());
+            } else {
+              std::visit([&](auto& tupl) -> void {
+                std::apply([&](auto&... args) -> void {
+                  std::execution::set_value((Receiver&&)state.receiver_, std::move(args)...);
+                }, tupl);
+              }, state.data_);
```

Comment:
This could also use `bulk_shared_state::apply`, right?

Reply:
Good point, done!